### ISSUE TITLE
[charts] Remove `series[].data[].id` from `scatter` series

### ISSUE THREAD

**Issue by JCQuintas** [mui/mui-x#15731]
We could remove the requirement for `id` from the scatter series `data` and make it optional.

We can use `{seriesId}-{dataIndex}` instead to the same effect.

We should test if there big performance issues and do some testing. It also might be important to tell the user of any side-effects.

**Search keywords**:

**Comment by alexfauquette** [comment]
That assumes the data can not be modified.

If you add/remove/update item the id is the only way to know what is happening. It's like the rule "don't use index to for the `key`". But it can be a fallback if needed

**Comment by JCQuintas** [comment]
True, but is there an advantage for scatter charts that bar/line chart don't need? Technically the issue is that they are in an array, and all our data use arrays.

**Comment by alexfauquette** [comment]
For bar/line chart, the data is an array where the index is used to do the matching with the axis data.
Technically the item id for bar/line should be the axes values since they are the only one that we assume to be unique.

That's the main difference between a line chart and a scatter chart with some linked points.

**Comment by JCQuintas** [comment]
Done in #16389, which made `id` optional on `ScatterValueType` (and `PieValueType`). It is now `id?: string | number`, so you no longer have to provide it.
